
This makes UML load at 0x8048000 rather than 0xa0000000 when 
CONFIG_MODE_SKAS is on and CONFIG_MODE_TT is off.  This will make it 
more valgrind-friendly, and also allow much greater physical memory 
sizes without needing to use highmem.

Signed-off-by: Paolo 'Blaisorblade' Giarrusso <blaisorblade_spam@yahoo.it>
---

 linux-2.6.9-current-paolo/arch/um/Makefile      |    6 +++++-
 linux-2.6.9-current-paolo/arch/um/Makefile-i386 |    6 ++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff -puN arch/um/Makefile~uml-load-low arch/um/Makefile
--- linux-2.6.9-current/arch/um/Makefile~uml-load-low	2004-09-19 18:27:20.918858848 +0200
+++ linux-2.6.9-current-paolo/arch/um/Makefile	2004-09-19 18:27:20.921858392 +0200
@@ -111,8 +111,12 @@ CPP_MODE_TT := $(shell [ "$(CONFIG_MODE_
 CONFIG_KERNEL_STACK_ORDER ?= 2
 STACK_SIZE := $(shell echo $$[ 4096 * (1 << $(CONFIG_KERNEL_STACK_ORDER)) ] )
 
+ifndef START
+  START = $$(($(TOP_ADDR) - $(SIZE)))
+endif
+
 CPPFLAGS_vmlinux.lds = $(shell echo -U$(SUBARCH) \
-	-DSTART=$$(($(TOP_ADDR) - $(SIZE))) -DELF_ARCH=$(ELF_ARCH) \
+	-DSTART=$(START) -DELF_ARCH=$(ELF_ARCH) \
 	-DELF_FORMAT=\"$(ELF_FORMAT)\" $(CPP_MODE_TT) \
 	-DKERNEL_STACK_SIZE=$(STACK_SIZE))
 
diff -puN arch/um/Makefile-i386~uml-load-low arch/um/Makefile-i386
--- linux-2.6.9-current/arch/um/Makefile-i386~uml-load-low	2004-09-19 18:27:20.919858696 +0200
+++ linux-2.6.9-current-paolo/arch/um/Makefile-i386	2004-09-19 18:27:20.921858392 +0200
@@ -4,6 +4,12 @@ else
 TOP_ADDR = 0xc0000000
 endif
 
+ifeq ($(CONFIG_MODE_SKAS),y)
+  ifneq ($(CONFIG_MODE_TT),y)
+     START = 0x8048000
+  endif
+endif
+
 CFLAGS += -U__$(SUBARCH)__ -U$(SUBARCH)
 
 ifneq ($(CONFIG_GPROF),y)
_
